src/Orchard.Web/Modules/Orchard.Widgets/Filters/WidgetFilter.cs

```diff
@@ -51,25 +49,7 @@ public class WidgetFilter : FilterProvider, IResultFilter {
                 return;
             }
 
-            // Once the Rule Engine is done:
-            // Get Layers and filter by zone and rule
-            // NOTE: .ForType("Layer") is faster than .Query<LayerPart, LayerPartRecord>()
-            IEnumerable<LayerPart> activeLayers = _orchardServices.ContentManager.Query<LayerPart>().ForType("Layer").List();
-
-            var activeLayerIds = new List<int>();
-            foreach (var activeLayer in activeLayers) {
-                // ignore the rule if it fails to execute
-                try {
-                    if (_ruleManager.Matches(activeLayer.LayerRule)) {
-                        activeLayerIds.Add(activeLayer.ContentItem.Id);
-                    }
-                }
-                catch(Exception e) {
-                    Logger.Warning(e, T("An error occured during layer evaluation on: {0}", activeLayer.Name).Text);
-                }
-            }
-
-            IEnumerable<WidgetPart> widgetParts = _widgetsService.GetWidgets(layerIds: activeLayerIds.ToArray());
+            var widgetParts = _widgetsService.GetWidgets(_layerEvaluationService.GetActiveLayerIds().ToArray());

```

Comment:
Personal preference maybe but I think it's nicer to maintain a pattern across the codebase where consumers of the API to continue to treat the `IEnumerable<T>` as `IEnumerable<T>` and trust the API to return the simplest underlying form possible. So in this case, trust LayerEvaluationService to return an array... [and see below]